ext/pcre/tests/preg_replace_callback_array_trampoline.phpt

```diff
@@ -0,0 +1,111 @@
+--TEST--
+preg_replace_callback_array() with a trampoline
+--FILE--
+<?php
+
+class TrampolineTest {
+    public function __call(string $name, array $arguments): string {
+        echo 'Trampoline for ', $name, PHP_EOL;
+        if ($name === 'trampolineThrow') {
+            throw new Exception('boo');
+        }
+        return "'" . $arguments[0][0] . "'";
+    }
+}
+$o = new TrampolineTest();
+$callback = [$o, 'trampoline'];
+$callbackThrow = [$o, 'trampolineThrow'];
+
+$regexesToTest = [
+    [
+        '@\b\w{1,2}\b@' => $callback,
+        '~\A.~' => $callback,
+    ],
+    [
+        '@\b\w{1,2}\b@' => $callback,
+        '~\A.~' => $trampolineThrow,
+    ],
+    [
+        '@\b\w{1,2}\b@' => $callback,
+        '~\A.~' => new stdClass(),
+    ],
+];
+
+$subjectsToTest = [
+    [
+        'a b3 bcd',
+        'v' => 'aksfjk',
+        12 => 'aa bb',
+        ['xyz'],
+    ],
+    'a b3 bcd',
+    [
+        new stdClass(),
+    ],
+    new stdClass(),
+];
+
+foreach ($regexesToTest as $regex) {
+    foreach ($subjectsToTest as $subject) {
+        try {
+            $matches = preg_replace_callback_array($regex, $subject);
+            var_dump($matches);
+        } catch (Throwable $e) {
+            echo $e::class, ': ', $e->getMessage(), PHP_EOL;
+        }
+    }
+}
+
+?>
+--EXPECTF--
+Warning: Undefined variable $trampolineThrow in %s on line %d
+Trampoline for trampoline
+Trampoline for trampoline
+Trampoline for trampoline
+Trampoline for trampoline
+
+Warning: Array to string conversion in %s on line %d
```

Comment:
Is this intentional?

Reply:
Yes the current code does do array to string conversions, so I wanted to preserve this.